test/tests/functional/pbs_resv_end_hook.py

```diff
@@ -0,0 +1,605 @@
+
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, attrs, id=rid)
+
+        self.server.delete(rid)
+        msg = 'Svr;Server@%s;PBS server internal error (15011) in Error ' \
+              'evaluating Python script, attribute '"'resources_used'"' is ' \
+              'part of a readonly object' % self.server.hostname.lower()
+        self.server.log_match(msg, tail=True, max_attempts=30, interval=2)
+
+    def test_delete_resv_occurrence(self):
+        """
+        Testcase to submit and confirm a standing reservation for two
+        occurrences, wait for the first occurrence to end and verify
+        the end hook for the same, delete the second occurrence and
+        verify the resvend hook for the latter.
+        """
+        self.server.import_hook(self.hook_name,
+                                TestResvEndHook.standing_resv_hook_script)
+
+        offset = 10
+        duration = 30
+        rid = self.submit_resv(offset, duration, rrule='FREQ=MINUTELY;COUNT=2')
+
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, attrs, id=rid)
+
+        attrs['reserve_state'] = (MATCH_RE, 'RESV_RUNNING|5')
+        self.server.expect(RESV, attrs, id=rid, offset=10)
+
+        self.logger.info('wait till 30 seconds until the reservation ends')
+        time.sleep(30)
+
+        msg = 'Hook;Server@%s;Reservation occurrence - 1' % \
+              self.server.hostname.lower()
+        self.server.log_match(msg, tail=True, interval=2, max_attempts=30)
+        self.logger.info('Reservation end hook ran for first occurrence of '
+                         'a standing reservation')
+
+        self.logger.info(
+            'wait for 10 seconds till the next occurrence is submitted')
+        time.sleep(10)
+
+        self.server.delete(rid)
+        msg = 'Hook;Server@%s;Reservation occurrence - 2' % \
+              self.server.hostname.lower()
+        self.server.log_match(msg, tail=True, interval=2, max_attempts=30)
+
+    def test_end_resv_occurrences(self):
+        """
+        Testcase to submit and confirm a standing reservation for two
+        occurrences, wait for the first occurrence to end and verify
+        the end hook for the same, wait for the second occurrence to
+        start and end, verify the resvend hook for the latter.
+        """
+        self.server.import_hook(self.hook_name,
+                                TestResvEndHook.standing_resv_hook_script)
+
+        offset = 10
+        duration = 30
+        rid = self.submit_resv(offset, duration, rrule='FREQ=MINUTELY;COUNT=2')
+
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, attrs, id=rid)
+
+        attrs['reserve_state'] = (MATCH_RE, 'RESV_RUNNING|5')
+        self.server.expect(RESV, attrs, id=rid, offset=10)
+
+        self.logger.info('Sleep for 30 seconds for the reservation occurrence '
```

Comment:
I'd suggest that you increase the timeout on test cases (by default it is 180 secs) where the test sleeps and then you have a logmatch which could also potentially go into a minute long lookup.
This would apply to all similar test cases